src/skin.cpp

```diff
@@ -245,16 +245,11 @@ void SkinInfillAreaComputation::applySkinExpansion(const Polygons& original_outl
  *
  * this function may only read/write the skin and infill from the *current* layer.
  */
-void SkinInfillAreaComputation::generateSkinInsets(SliceLayerPart* part)
+void SkinInfillAreaComputation::generateSkinInsetsAndInnerSkinInfill(SliceLayerPart* part)
```

Comment:
What this function really does is:
* Generate insets for skin (lines 269 to 288). The result gets stored by reference in `skin_part`.
* Generate the skin infill area by taking the last inset in the `skin_part` and insetting it a bit more (lines 289 and 290).

There's an edge case at the top (lines 264 through 268) that could also be done in the second part, generating the skin infill areas.